gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java

```diff
@@ -102,6 +122,94 @@ public GroovyCodeEditor(Scene scene) {
         codeArea.setOnDragOver(this::onDragOver);
         codeArea.setOnDragDropped(this::onDragDropped);
         codeArea.setOnSelectionDrag(p -> allowedDrag = true);
+
+        codeArea.caretListener = (observable, oldvalue, newvalue) -> {
+            bracketsMatches.searchingForMatches = false;
+            codeArea.setStyleSpans(0, computeHighlighting(codeArea.getText()));
+            if (newvalue > 0) {
+                codeArea.caretPositonText = codeArea.getText(newvalue - 1, newvalue);
+                findBracketsMatches(codeArea.getText());
+            }
+        };
+
+        codeArea.caretPositionProperty().addListener(codeArea.caretListener);
+
+        codeProperty().addListener((observable, oldvalue, newvalue) -> bracketsMatches.searchingForMatches = false);
+    }
+
+    private void findBracketsMatches(String text) {
+        if (codeArea.caretPositonText  != null) {
+            if (codeArea.caretPositonText .equals(Character.toString(')'))) {
+                bracketsMatches.leftMatch = true;
```

Comment:
I’m not sure that leftMatch is a good name. It could be clever to use a integer equal to 1 (forward) and -1 (backward). Add this integer to the current position to move forward or backward to try to find the matching brackets